src/storage/admin/AdminTaskManager.cpp

```diff
@@ -25,10 +26,75 @@ bool AdminTaskManager::init() {
 
   bgThread_->addTask(&AdminTaskManager::schedule, this);
   shutdown_ = false;
+  handleUnreportedTasks();
   LOG(INFO) << "exit AdminTaskManager::init()";
   return true;
 }
 
+void AdminTaskManager::handleUnreportedTasks() {
+  if (env_ == nullptr) return;
+  std::unique_ptr<kvstore::KVIterator> iter;
+  auto kvRet = env_->adminStore_->scan(&iter);
+  if (kvRet != nebula::cpp2::ErrorCode::SUCCEEDED || iter == nullptr) return;
+  std::vector<std::string> keys;
+  for (; iter->valid(); iter->next()) {
+    folly::StringPiece key = iter->key();
+    JobID jobId = *reinterpret_cast<const JobID*>(key.data());
+    TaskID taskId = *reinterpret_cast<const TaskID*>(key.data() + sizeof(JobID));
+    keys.emplace_back(key.data(), key.size());
+    folly::StringPiece val = iter->val();
+    nebula::meta::cpp2::JobStatus jobStatus =
+        *reinterpret_cast<const nebula::meta::cpp2::JobStatus*>(val.data() +
+                                                                sizeof(meta::cpp2::AdminCmd));
+    nebula::cpp2::ErrorCode errCode;
+    nebula::meta::cpp2::StatsItem statsItem;
+    if (jobStatus != nebula::meta::cpp2::JobStatus::FINISHED) {
+      errCode = nebula::cpp2::ErrorCode::E_TASK_EXECUTION_FAILED;
+      statsItem.set_status(nebula::meta::cpp2::JobStatus::FAILED);
+    } else {
+      errCode = nebula::cpp2::ErrorCode::SUCCEEDED;
+      statsItem.set_status(nebula::meta::cpp2::JobStatus::FINISHED);
```

Comment:
Maybe always set FAILED in statsItem?

Reply:
Besides, there are no code of setting job status to `FINISHED`, do I miss anything? There should not be the `else` branch?